pkg/admissioncontroller/apis/config/v1alpha1/types.go

```diff
@@ -33,6 +33,8 @@ type AdmissionControllerConfiguration struct {
 	LogLevel string `json:"logLevel"`
 	// Server defines the configuration of the HTTP server.
 	Server ServerConfiguration `json:"server"`
+	// Debugging holds configuration for Debugging related features.
+	Debugging componentbaseconfigv1alpha1.DebuggingConfiguration `json:"debugging"`
```

Comment:
Shouldn't these fields be optional (pointer type, optional + omitempty tag)? 

(same for other API changes)

Reply:
I would propose to make them optional only for `gardenlet` where other configuration options on this level are also optional. For all other components, there are no optional configuration fields at all, so introducing one now would be quite inconsistent.

Reply:
Yeah, keeping it as is probably has little/no impact, still, I'm looking at clarification whether we want to follow the typical API conventions for the configuration API groups as well. If yes then we have to make it optional. What is our general approach here?

Reply:
We don't seem to have a general approach at the moment - `gardenlet` and the rest are already different. I would vote to keep them consistent, so make it optional only for `gardenlet`.

Reply:
When adding the fields, I was also torn whether to add pointer, optional and omitempty tag.
I didn't do it because:
- the field/struct is somewhat optional: leaving it out in YAML is valid and leads to the default values in the internal representation (false, false) as expected
- later on, when we want to make it enabled by default, we can still distinguish between unset and explicitly disabled (the members (`debugging.{enableProfiling,enableContentionProfiling}`) are pointer+omitempty fields. We can simply call [`RecommendedDebuggingConfiguration`](https://github.com/gardener/gardener/blob/06242c99ecebf7442021acbb14a895eca27f7f78/vendor/k8s.io/component-base/config/v1alpha1/defaults.go#L86) then.
- adding a pointer in the external version forces me to add conversion code (in which I see no point). The generator complains with 
```
E0903 17:00:43.753112   10045 conversion.go:755] Warning: could not find nor generate a final Conversion function for github.com/gardener/gardener/pkg/scheduler/apis/config/v1alpha1.SchedulerConfiguration -> github.com/gardener/gardener/pkg/scheduler/apis/config.SchedulerConfiguration
E0903 17:00:43.753255   10045 conversion.go:756]   the following fields need manual conversion:
E0903 17:00:43.753279   10045 conversion.go:758]       - Debugging
```
and adds the following comment in the generated code
```go
// WARNING: in.Debugging requires manual conversion: inconvertible types (*k8s.io/component-base/config/v1alpha1.DebuggingConfiguration vs k8s.io/component-base/config.DebuggingConfiguration)

```
- it was easy to implement like this as I have handled it consistently across all components
- it seemed like we didn't care about API conventions in our config APIs so far and they are quite a mess in several other ways as well (see https://github.com/gardener/gardener/issues/4380 and https://github.com/gardener/gardener/issues/4605)

At the end, this was just the path of least resistance and I didn't care about it too much.

Like you, I don't think we have a general approach right now. I think, we should probably come to an agreement on this some time and document it properly. Then, we should take the guideline and consistently apply it to all config APIs to clean everything up (build a new `v1beta1` version?)

As for this PR: There seems to be no right or wrong here and the APIs have already been inconsistent before my change.
I would prefer to just leave it like this. I just don't see the point or benefit in changing it at all.
But I can change it, if you insist.

Reply:
I can only agree with @timebertt. Those were also the reasons I didn't reject it in my review. Thanks for elaborating.

Reply:
Ok great, seems like everyone agrees. Shall we then merge the PR now?